nixos/modules/programs/opengamepadui.nix

```diff
@@ -0,0 +1,266 @@
+      });
+in
+{
+  options.programs.opengamepadui = {
+    enable = lib.mkEnableOption "opengamepadui";
+
+    args = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      description = ''
+        Arguments to be passed to OpenGamepadUI
+      '';
+    };
+
+    package = lib.mkPackageOption pkgs "OpenGamepadUI" {
+      default = [ "opengamepadui" ];
+    };
+
+    extraPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      example = lib.literalExpression ''
+        with pkgs; [
+          gamescope
+        ]
+      '';
+      description = ''
+        Additional packages to add to the OpenGamepadUI environment.
+      '';
+    };
+
+    fontPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = config.fonts.packages;
+      defaultText = lib.literalExpression "builtins.filter lib.types.package.check config.fonts.packages";
+      example = lib.literalExpression "with pkgs; [ source-han-sans ]";
+      description = ''
+        Font packages to use in OpenGamepadUI.
+
+        Defaults to system fonts, but could be overridden to use other fonts — useful for users who would like to customize CJK fonts used in opengamepadui. According to the [upstream issue](https://github.com/ValveSoftware/opengamepadui-for-linux/issues/10422#issuecomment-1944396010), opengamepadui only follows the per-user fontconfig configuration.
+      '';
+    };
+
+    gamescopeSession = lib.mkOption {
+      description = "Run a GameScope driven OpenGamepadUI session from your display-manager";
+      default = { };
+      type = lib.types.submodule {
+        options = {
+          enable = lib.mkEnableOption "GameScope Session";
+          args = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [
+              "--prefer-output"
+              "*,eDP-1"
+              "--xwayland-count"
+              "2"
+              "--default-touch-mode"
+              "4"
+              "--hide-cursor-delay"
+              "3000"
+              "--fade-out-duration"
+              "200"
+              "--steam"
+            ];
+            description = ''
+              Arguments to be passed to GameScope for the session.
+            '';
+          };
+
+          env = lib.mkOption {
+            type = lib.types.attrsOf lib.types.str;
+            default = { };
+            description = ''
+              Environmental variables to be passed to GameScope for the session.
+            '';
+          };
+        };
+      };
+    };
+
+    inputplumber.enable = lib.mkEnableOption ''
+      Run InputPlumber service for input management and gamepad configuration.
+    '';
+  };
+
+  config = lib.mkIf cfg.enable {
+    hardware.graphics = {
+      # this fixes the "glXChooseVisual failed" bug, context: https://github.com/NixOS/nixpkgs/issues/47932
+      enable = true;
+      enable32Bit = true;
```

Comment:
This prevents use on aarch64-linux. Is it absolutely needed?

Reply:
https://github.com/NixOS/nixpkgs/blob/bffc22eb12172e6db3c5dde9e3e5628f8e3e7912/nixos/modules/hardware/graphics.nix#L103

Reply:
Then about

```suggestion
      enable32Bit = pkgs.stdenv.hostPlatform.isx86_64;
```

Reply:
While this OpenGamepadUI derivation won't support `aarch64-linux` right now, I'll add this suggestion so it will be ready when we can support that architecture.